privx_api/privx_api.py

```diff
@@ -250,6 +251,21 @@ def _http_put(self, urlname: str,
 
         return conn.getresponse()
 
+    def _http_delete(self, urlname: str,
+                  body: dict = {},
+                  path_params: dict = {},
+                  query_params: dict = {}) -> http.client.HTTPResponse:
```

Comment:
Using mutable types as function default arguments is an anti-pattern every Python programmer has to encounter once. The mutable default arguments are initialized once, and if the values are mutated inside the method, these 'bad' values will be present when the method is next time called. 

The _get_headers() method uses similar antipattern.

If the values are not mutated, no harm is done, but the code is still prone to break easily. 

Demonstration: 
```
>>> def foo(y, x = []):
...  x.append(y)
...  return x
... 
>>> foo(1)
[1]
>>> foo(1)
[1, 1]
```

Reply:
Thanks for reporting the issue, I tried fixing it, please review and share feedback. I am not programmer, just writing small python scripts or modifying existing once.